digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala

```diff
@@ -1,44 +1,209 @@
+
+  override def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  val trafficSignService: TrafficSignService = Digiroad2Context.trafficSignService
+
+  private val supportedTrafficSigns = Set[TRTrafficSignType](TRTrafficSignType.SpeedLimit, TRTrafficSignType.EndSpeedLimit, TRTrafficSignType.SpeedLimitZone, TRTrafficSignType.EndSpeedLimitZone,
+    TRTrafficSignType.UrbanArea, TRTrafficSignType.EndUrbanArea, TRTrafficSignType.PedestrianCrossing, TRTrafficSignType.MaximumLength, TRTrafficSignType.Warning,
+    TRTrafficSignType.NoLeftTurn, TRTrafficSignType.NoRightTurn, TRTrafficSignType.NoUTurn, TRTrafficSignType.ClosedToAllVehicles, TRTrafficSignType.NoPowerDrivenVehicles,
+    TRTrafficSignType.NoLorriesAndVans, TRTrafficSignType.NoVehicleCombinations, TRTrafficSignType.NoAgriculturalVehicles, TRTrafficSignType.NoMotorCycles, TRTrafficSignType.NoMotorSledges,
+    TRTrafficSignType.NoVehiclesWithDangerGoods, TRTrafficSignType.NoBuses, TRTrafficSignType.NoMopeds, TRTrafficSignType.NoCyclesOrMopeds, TRTrafficSignType.NoPedestrians,
+    TRTrafficSignType.NoPedestriansCyclesMopeds, TRTrafficSignType.NoRidersOnHorseback, TRTrafficSignType.NoEntry, TRTrafficSignType.OvertakingProhibited, TRTrafficSignType.EndProhibitionOfOvertaking,
+    TRTrafficSignType.MaxWidthExceeding, TRTrafficSignType.MaxHeightExceeding, TRTrafficSignType.MaxLadenExceeding, TRTrafficSignType.MaxMassCombineVehiclesExceeding, TRTrafficSignType.MaxTonsOneAxleExceeding,
+    TRTrafficSignType.MaxTonsOnBogieExceeding, TRTrafficSignType.WRightBend, TRTrafficSignType.WLeftBend, TRTrafficSignType.WSeveralBendsRight, TRTrafficSignType.WSeveralBendsLeft,
+    TRTrafficSignType.WDangerousDescent, TRTrafficSignType.WSteepAscent, TRTrafficSignType.WUnevenRoad, TRTrafficSignType.WChildren)
+
+  private val longValueFieldMappings = Map(
+    "koordinaatti x" -> "lon",
+    "koordinaatti y" -> "lat"
+  )
+
+  private val nonMandatoryMappings = Map(
+    "arvo" -> "value",
+    "kaksipuolinen merkki" -> "twoSided",
+    "liikennevirran suunta" -> "trafficDirection",
+    "suuntima" -> "bearing"
+  )
+
+  private val codeValueFieldMappings = Map(
+    "liikennemerkin tyyppi" -> "trafficSignType"
+  )
+
+  val mappings = longValueFieldMappings ++ nonMandatoryMappings ++ codeValueFieldMappings
+
+  private val mandatoryFields = List("koordinaatti x", "koordinaatti y", "liikennemerkin tyyppi")
+
+  val MandatoryParameters: Set[String] = mappings.keySet ++ mandatoryFields
+
+  private def findMissingParameters(csvRowWithHeaders: Map[String, String]): List[String] = {
+    MandatoryParameters.diff(csvRowWithHeaders.keys.toSet).toList
+  }
+
+  def rowToString(csvRowWithHeaders: Map[String, Any]): String = {
+    csvRowWithHeaders.view map { case (key, value) => key + ": '" + value + "'" } mkString ", "
+  }
+
+  private def verifyDoubleType(parameterName: String, parameterValue: String): ParsedAssetRow = {
+    if(parameterValue.matches("[0-9.]*")) {
```

Comment:
Could it be replaced by _isDigit?

Reply:
Not if the value contains a decimal point